bipentropy/bipentropy.py

```diff
@@ -21,12 +21,24 @@
 
 import hmac
 import hashlib
-import base58
 from mnemonic import Mnemonic as bip39
 from pycoin.symbols.btc import network as BTC
+from pycoin.encoding.bytes32 import from_bytes_32, to_bytes_32
 
 
 class BIPEntropy(object):
+
+    def __get_k_from_node(self, node):
+        return to_bytes_32(node.secret_exponent())
+
+    def __derive_k(self, path, xprv):
+        path = path.replace("m/", "").replace("'", "p")
+        node = xprv.subkey_for_path(path)
+        return self.__get_k_from_node(node)
+
+    def __hmac_sha512(self, message_k):
+        return hmac.new(message_k, msg=b'bip-entropy-from-k', digestmod=hashlib.sha512).digest()
```

Comment:
This is wrong. The format of the python function is `hmac.new(key, msg, digest)`, where `key` is a shared key or salt, and message is the payload you are hashing like a password etc. See the python reference https://docs.python.org/3/library/hmac.html#hmac.new

And examples in the wild in [pycoin](https://github.com/richardkiss/pycoin/blob/de89fa4ffb5b7eb518c9ab2859c9d767f6b3e31d/pycoin/key/BIP32Node.py#L43), [python-mnemonic](https://github.com/trezor/python-mnemonic/blob/master/mnemonic/mnemonic.py#L258), and [Bitcoin Core](https://github.com/bitcoin/bitcoin/blob/267793af8b03e2a11af8a51d7091495bbd065b62/test/util/rpcauth-test.py#L41-L42).

Reply:
Yes, okay.